### PR TITLE
Move the --farm flag to farm build command

### LINKED ISSUE #20752: docs: podman farm: --farm option should be moved
```console
$ bin/podman farm --help
...
Available Commands:
  build       Build a container image for multiple architectures
  create      Create a new farm
  list        List all existing farms
  remove      Remove one or more farms
  update      Update an existing farm

Options:
  -f, --farm string   Farm to use for builds (default "myfarm")
```
The `-f/--farm` option must be moved to `podman farm build`.

### PR DESCRIPTION
**Pull request by cgiradkar**
closes #20752

#### Does this PR introduce a user-facing change?

```release-note
The --farm flag has been moved to farm build command from farm command
```

### CODE REVIEW

**Comment by jakecorrenti** [comment]
If you don't have any new tests to add you need to have the [NO NEW TESTS] line in your commit. Can you also add a more descriptive commit message?

**Comment by rhatdan** [comment]
@cgiradkar why is this draft?  You should only use Draft if you want your PR to run through QE but not be merged.  If you would like to have a PR reviewed and merged if it passes tests, do not set it to draft.

File: cmd/podman/farm/build.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ type buildOptions struct {
 	buildOptions common.BuildFlagsWrapper
 	local        bool
 	platforms    []string
+	Farm         string
```
Comment (umohnani8) [inline cmd/podman/farm/build.go:22]: This can be private, please switch it to `farm`.

**Comment by umohnani8** [comment]
@cgiradkar nice work! You need to make changes to the docs as well

**Pull request merged**